### PR TITLE
Fix Ruby 3 compatibility issues in Connection::Client#pause

### PR DESCRIPTION
**Pull request by MmKolodziej**
Aims to tackle https://github.com/karafka/karafka/issues/699

### CODE REVIEW

**Comment by mensfeld** [comment]
@MmKolodziej thank you. Is it sill a WIP?

File: lib/karafka/connection/api_adapter.rb [hunk 1]
```diff
@@ -84,17 +84,16 @@ def subscribe(topic)
...
         # @param consumer_group [Karafka::Routing::ConsumerGroup] consumer group details
-        # @return [Array] array with all the details required to pause kafka consumer
+        # @return [Hash] hash with all the details required to pause kafka consumer
         def pause(topic, partition, consumer_group)
-          [
-            Karafka::App.config.topic_mapper.outgoing(topic),
-            partition,
-            {
+          {
+            args: [Karafka::App.config.topic_mapper.outgoing(topic), partition],
```
Comment (MmKolodziej) [inline lib/karafka/connection/api_adapter.rb:90]: Client could simply do something like `*args[0..-2], **args.last`, but I think that if there's an adapter, it should try and be more helpful with how to pass the params

**Comment by MmKolodziej** [comment]
> @MmKolodziej thank you. Is it sill a WIP?

Ready now, wanted to see whether CI passes before converting it from a draft. :)

File: lib/karafka/connection/api_adapter.rb (hunk 1 above)
Comment (mensfeld) [reply lib/karafka/connection/api_adapter.rb:90]: I agree :)

**Review by mensfeld — APPROVED** [review]

**Comment by mensfeld** [comment]
Looks good. Do you want me to craft a release with this?

**Comment by MmKolodziej** [comment]
> Looks good. Do you want me to craft a release with this?

If you could, please. This is the last thing that's popping up in production environments regarding keyword arguments deprecations.

**Comment by mensfeld** [comment]
@MmKolodziej released.

**Comment by MmKolodziej** [comment]
Hm, the release is there, but tags look a little bit weird: https://github.com/karafka/karafka/tags

**Comment by mensfeld** [comment]
I think I fixed it. No idea why that happened.

**Comment by MmKolodziej** [comment]
Cool, thanks! :tada:

**Pull request merged**